Assets/Scripts/Singletons/DataModelSingletons/HangarManager.cs

```diff
@@ -9,9 +9,12 @@ public class HangarManager : MonoBehaviour
     [SerializeField] private GameObject shipInstancePrefab;
 
     public static HangarSlot[] HangarSlots { get; private set; }
-    public static Battery[] Batteries { get; private set; }
+    public static BatteryInteractable[] BatteryInteractables { get; private set; }
```

Comment:
The logic is separated out well now, but I don't like how we have two separate arrays here, each element of which relates to the same individual object. 

1. We have to call ``Find`` twice 
2. How can we guarantee that we have an even amount of elements in total, i.e. the same number of ``BatteryInteractable``s as ``BatteryCharging``s? If they don't match up, then we will have a lot of problems.
3. It makes it harder to do operations over them both at the same time. 

Personally I think we should have a script that wraps the two, maybe just called ``Battery``. Then we can just ``Find`` the Batteries and access both the Charging and Interactable code through them. Makes them easier to manage.

We could then have a ``BatterySaveData`` struct on that wrapper class. Which could load both position and charging data. Having two arrays and two save data structs makes it pretty awkward, as we have to loop through both arrays and hope that they are of the same length. We already check if the json file has more elements than the Battery, which is unlikely to happen anyway at the moment, but it'll be harder if there's two separate arrays. Before, there was just one.

This should make the two components on the wrapper available while keeping the internal logic of both separate (which you have already done). 

Something like: 
```
public BatteryInteractable BatteryInteractable;
public BatteryCharging BatteryCharging;

public BatterySaveData saveData;
...
```

Having a wrapper class also means that we can do ``[RequireComponent(typeof(BatteryInteractable))]``, etc to make sure the required 2 dependencies are there. 

Because both the components you made are on the Container object on the prefab, this shouldn't affect the physics loop. What do you think?